fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheAnalyzer.java

```diff
@@ -0,0 +1,450 @@
+                                               List<CompoundPredicate> compoundPredicates) {
+        getPartitionKeyFromWhereClause(stmt.getWhereClause(), partColumn, compoundPredicates);
+        List<TableRef> tableRefs = stmt.getTableRefs();
+        for (TableRef tblRef : tableRefs) {
+            if (tblRef instanceof InlineViewRef) {
+                InlineViewRef viewRef = (InlineViewRef) tblRef;
+                QueryStmt queryStmt = viewRef.getViewStmt();
+                if (queryStmt instanceof SelectStmt) {
+                    getPartitionKeyFromSelectStmt((SelectStmt) queryStmt, partColumn, compoundPredicates);
+                }
+            }
+        }
+    }
+
+    /**
+     * Only support case 1
+     * 1.key >= a and key <= b
+     * 2.key = a or key = b
+     * 3.key in(a,b,c)
+     */
+    private void getPartitionKeyFromWhereClause(Expr expr, Column partColumn,
+                                                List<CompoundPredicate> compoundPredicates) {
+        if (expr == null) {
+            return;
+        }
+        if (expr instanceof CompoundPredicate) {
+            CompoundPredicate cp = (CompoundPredicate) expr;
+            if (cp.getOp() == CompoundPredicate.Operator.AND) {
+                if (cp.getChildren().size() == 2 && cp.getChild(0) instanceof BinaryPredicate &&
+                        cp.getChild(1) instanceof BinaryPredicate) {
+                    BinaryPredicate leftPre = (BinaryPredicate) cp.getChild(0);
+                    BinaryPredicate rightPre = (BinaryPredicate) cp.getChild(1);
+                    String leftColumn = getColumnName(leftPre);
+                    String rightColumn = getColumnName(rightPre);
+                    if (leftColumn.equalsIgnoreCase(partColumn.getName()) &&
+                            rightColumn.equalsIgnoreCase(partColumn.getName())) {
+                        compoundPredicates.add(cp);
+                    }
+                }
+            }
+            for (Expr subExpr : expr.getChildren()) {
+                getPartitionKeyFromWhereClause(subExpr, partColumn, compoundPredicates);
+            }
+        }
+    }
+
+    private String getColumnName(BinaryPredicate predicate) {
+        SlotRef slot = null;
+        if (predicate.getChild(0) instanceof SlotRef) {
+            slot = (SlotRef) predicate.getChild(0);
+        } else if (predicate.getChild(0) instanceof CastExpr) {
+            CastExpr expr = (CastExpr) predicate.getChild(0);
+            if (expr.getChild(0) instanceof SlotRef) {
+                slot = (SlotRef) expr.getChild(0);
+            }
+        }
+
+        if (slot != null) {
+            return slot.getColumnName();
+        }
+        return "";
+    }
+
+    /**
+     * Check the selectStmt and tableRefs always group by partition key
+     * 1. At least one group by
+     * 2. group by must contain partition key
+     * 3. AggregateInfo cannot be distinct agg
```

Comment:
Why AggregateInfo cannot be distinct agg? if distinct cache result is final result, I think which is OK.

Reply:
Yes, It's support COUNT(DISTINCT xxx)